compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -1159,7 +1159,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             // those that do.
             self.one_bound_for_assoc_type(
                 || traits::supertraits(tcx, trait_ref),
-                trait_ref.print_only_trait_path(),
+                || tcx.def_path_str(trait_ref.def_id()),
```

Comment:
this can use print_only_trait_name instead of introducing a closure here, i think